### ISSUE TITLE
FindAllReferences on KeyValuePair.Deconstruct extension method finds no usages

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#24184]
But it is actually used.
If I comment out this method and re-compile, I get an error:
`1>DiagnosticAnalyzer\AnalysisState.cs(399,47,399,64): error CS1061: 'KeyValuePair<DiagnosticAnalyzer, int>' does not contain a definition for 'Deconstruct' and no extension method 'Deconstruct' accepting a first argument of type 'KeyValuePair<DiagnosticAnalyzer, int>' could be found (are you missing a using directive or an assembly reference?)`

**Comment by jcouv** [comment]
I'm able to repro in test:
```vb
        <WpfFact, Trait(Traits.Feature, Traits.Features.FindReferences)>
        <WorkItem(18963, "https://github.com/dotnet/roslyn/issues/18963")>
        Public Async Function FindReferences_DeconstructionExtension() As Task
            Dim input =
<Workspace>
    <Project Language="C#" CommonReferences="true">
        <Document><![CDATA[
static class Extension
{
    public static void {|Definition:Decons$$truct|}<TKey, TValue>(this KeyValuePair<TKey, TValue> c, out TKey x1, out TValue x2) { x1 = default; x2 = default; }
}
class C
{
    public void M()
    {
        var kvp = new KeyValuePair<int, string>(1, "hello");
        [|var (x1, x2)|] = kvp;
        foreach ([|var (y1, y2)|] in new[] { this }) { }
        [|(x1, (x2, _))|] = (1, this);
        (x1, x2) = (1, 2);
    }
}
        ]]>
        </Document>
    </Project>
</Workspace>
            Await TestAPIAndFeature(input)
        End Function
```

**Comment by CyrusNajmabadi** [comment]
@jcouv What did you miss when you wrote this? :)

**Comment by jcouv** [comment]
No clue yet ;-)

**Comment by jcouv** [comment]
Turns out my repro was buggy. Will investigate further.

**Comment by jcouv** [comment]
Queued a fix. The problem is that I wasn't looking in all documents, only one... :-S

**Comment by jcouv** [comment]
Fixed in 15.6 preview 4.
